qpc/insights/auth.py

```diff
@@ -0,0 +1,159 @@
+"""Insights Auth is used to handle the Insights authentication workflow."""
+
+import http
+import time
+from logging import getLogger
+
+import requests
+from requests.exceptions import BaseHTTPError, ConnectionError
+
+from qpc import messages
+from qpc.insights.exceptions import InsightsAuthError
+from qpc.translation import _
+from qpc.utils import CONFIG_SSO_HOST_KEY, read_insights_config
+
+logger = getLogger(__name__)
+
+DISCOVERY_CLIENT_ID = "discovery-client-id"
+INSIGHTS_REALM = "redhat-external"
+INSIGHTS_SCOPE = "api.console"
+GRANT_TYPE = "urn:ietf:params:oauth:grant-type:device_code"
+DEVICE_AUTH_ENDPOINT = (
+    f"/auth/realms/{INSIGHTS_REALM}/protocol/openid-connect/auth/device"
+)
+TOKEN_ENDPOINT = f"/auth/realms/{INSIGHTS_REALM}/protocol/openid-connect/token"
+
+
+class InsightsAuth:
+    """Implement the Insights Device Authorization workflow."""
+
+    def __init__(self):
+        self.auth_request = None
+        self.token_response = None
+        self.auth_token = None
+
+    def request_auth(self):
+        """Initialize a device authorization workflow request.
+
+        :returns: authorization request object
+        """
+        self.auth_request = None
+
+        config = read_insights_config()
+        insights_sso_server = config.get(CONFIG_SSO_HOST_KEY)
+        url = f"https://{insights_sso_server}{DEVICE_AUTH_ENDPOINT}"  # Always SSL
```

Comment:
Must it _always_ be SSL, though? I know in principal it should be, but what if someone is running a local development/test server?

Maybe we don't want to support that use case,  and maybe _always_ HTTPS/SSL is a fair expectation, but I wanted to raise the question here because I wasn't absolutely sure.

Reply:
While that's not enabled by default (HTTP), folks can enable it in Keycloak if they choose to for their dev environments.   Of course, they'll get hit by a lightning bolt from the Security demi-gods ⚡ 

Again, big No-No for production, so I'd prefer to keep ours https.  This is for Insights, even for testing, stage sso is https, customers would not be hitting their dev Keycloak for that anyways.  We can always add support for http in the future if anyone asks, highly doubtful.